completion.bash

```diff
@@ -0,0 +1,80 @@
+# -*- mode: sh; sh-shell: bash; -*-
+
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+_charmcraft()
+{
+    local cur prev words cword cmd cmds
+    cmds=(build version login logout whoami)
+    _init_completion || return
+
+    # only offer long options, as they should be self-explanatory (and
+    # it's not like it's more typing for the user)
+
+    if [ "$cword" -eq 1 ]; then
+        COMPREPLY=( $(compgen -W "--help --verbose --quiet ${cmds[*]}" -- "$cur") )
+        return
+    fi
+
+    case "$prev" in
+        --help)
+            return
+            ;;
+        --verbose|--quiet)
+            COMPREPLY=( $(compgen -W "${cmds[*]}" -- "$cur") )
+            return
+            ;;
+        login|logout|whoami|version)
+            COMPREPLY=( $(compgen -W "--help" -- "$cur") )
+            return
+            ;;
+        build)
+            COMPREPLY=( $(compgen -W "--help --from --entrypoint --requirement" -- "$cur") )
```

Comment:
Please make --entrypoint to filter by "*.py" and --requirement to filter by "*.txt", thanks!

Reply:
done :-)